Add more debug information for ANCM. (#9846)
aspnet/Docs: aspnetcore/host-and-deploy/iis/troubleshoot.md

This error occurs within the app's code during startup or while creating a response. The response may contain no content, or the response may appear as a *500 Internal Server Error* in the browser. The Application Event Log usually states that the app started normally. From the server's perspective, that's correct. The app did start, but it can't generate a valid response. [Run the app at a command prompt](#run-the-app-at-a-command-prompt) on the server or [enable the ASP.NET Core Module stdout log](#aspnet-core-module-stdout-log) to troubleshoot the problem.

### ErrorCode '0x800700c1'

```
EventID: 1010
Source: IIS AspNetCore Module V2
Failed to start application '/LM/W3SVC/6/ROOT/', ErrorCode '0x800700c1'.
```

> The app failed to start due to not being able to load the dll. 

Reviewer: ```suggestion
The app failed to start because the app's assembly (*.dll*) couldn't be loaded. 
```

> This error occurs means there is a bitness mismatch between the published application and the w3wp/iisexpress process. If publishing for x86, confirm that the Application Pool->Advanced Settings...->`Enable 32-Bit Applications` is set to true. If publishing for x64, confirm that `Enable 32-Bit Applications` is set to false.

Reviewer: ```suggestion
This error occurs when there's a bitness mismatch between the published app and the w3wp/iisexpress process.

Confirm that the app pool's 32-bit setting is correct:

1. Select the app pool in IIS Manager's **Application Pools**.
1. Select **Advanced Settings** under **Edit Application Pool** in the **Actions** panel.
1. Set **Enable 32-Bit Applications**:
   * If publishing for 32-bit (x86), set the value to `false`.
   * If publishing for 64-bit (x64), set the value to `true`.
```

### Connection reset

If an error occurs after the headers are sent, it's too late for the server to send a **500 Internal Server Error** when an error occurs. This often happens when an error occurs during the serialization of complex objects for a response. This type of error appears as a *connection reset* error on the client. [Application logging](xref:fundamentals/logging/index) can help troubleshoot these types of errors.

### Other behavior

> If for any other reason the application didn't start, add the following to your web.config. to enable debug logs.

Reviewer: This breaks the organization of the topic. There's a section below this for **Troubleshoot app startup errors** that this bit should be rolled into under a better heading. Would you like me to do it?

Author: Yeah go ahead, we are overloaded on our side right now 😄
